runtime/common/src/lib.rs

```diff
@@ -62,7 +73,96 @@ macro_rules! decl_common_types {
+                StorageVersion::new(4).put::<CouncilMembership>();
+                StorageVersion::new(4).put::<TechnicalCommittee>();
+                StorageVersion::new(4).put::<TechnicalCommitteeMembership>();
+                StorageVersion::new(4).put::<Bounties>();
+                StorageVersion::new(1).put::<CampaignAssets>();
+                StorageVersion::new(1).put::<MarketAssets>();
+                StorageVersion::new(1).put::<CustomAssets>();
+                StorageVersion::new(15).put::<Contracts>();
+                log::info!("FixStorageVersion: Done!");
+                <Runtime as frame_system::Config>::DbWeight::get().writes(11)
+            }
+
+            #[cfg(feature = "try-runtime")]
+            fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+                Ok(vec![])
+            }
+
+            #[cfg(feature = "try-runtime")]
+            fn post_upgrade(_: Vec<u8>) -> Result<(), DispatchError> {
+                Ok(())
+            }
+        }
+
+        type TrieId = BoundedVec<u8, ConstU32<128>>;
+
+        /// Information for managing an account and its sub trie abstraction.
+        /// This is the required info to cache for an account.
+        #[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+        pub struct ContractInfo {
+            pub trie_id: TrieId,
+            pub code_hash: <Runtime as frame_system::Config>::Hash,
+            pub storage_bytes: u32,
+            pub storage_items: u32,
+            pub storage_byte_deposit: Balance,
+            pub storage_item_deposit: Balance,
+            pub storage_base_deposit: Balance,
+        }
+
+        struct ClearContractsChildTries;
+
+        impl OnRuntimeUpgrade for ClearContractsChildTries {
+            fn on_runtime_upgrade() -> frame_support::weights::Weight {
+                log::info!("ClearContractsChildTries: Starting...");
+                let mut total_writes = 0u64;
+                for (_, contract_info) in storage_key_iter::<AccountId, ContractInfo, Twox64Concat>(
+                    b"Contracts",
+                    b"ContractInfoOf",
+                ) {
+                    let trie_id = contract_info.trie_id;
+                    let inner_trie_id = trie_id.into_inner();
+                    let child_info = ChildInfo::new_default(&inner_trie_id);
+                    let multi_removal_result = child::clear_storage(&child_info, None, None);
+                    let writes = multi_removal_result.loops as u64;
+                    log::info!(
+                        "ClearContractsChildTries: Cleared trie {:?} in {:?} loops",
+                        inner_trie_id,
+                        writes
+                    );
+                    total_writes = total_writes.saturating_add(writes);
+                }
+                log::info!("ClearContractsChildTries: Done!");
+                <Runtime as frame_system::Config>::DbWeight::get().writes(total_writes)
```

Comment:
```suggestion
                <Runtime as frame_system::Config>::DbWeight::get().reads_writes(total_writes, total_writes)
```

Reads for the lookups?

Reply:
e95f9091e76e0feb02fb9d5540d363eebe7bac70